cloud/manifest_reader.cc

```diff
@@ -230,7 +230,9 @@ IOStatus ManifestReader::GetMaxFileNumberFromManifest(FileSystem* fs,
     }
     uint64_t f;
     if (edit.GetNextFileNumber(&f)) {
-      assert(*maxFileNumber <= f);
+      // Disabled temporarily.
+      // TODO: Reenable once the cloud manifest consistency issue is addressed.
+      // assert(*maxFileNumber <= f);
       *maxFileNumber = f;
```

Comment:
We could `std::max(*maxFileNumber, f);` here.

Reply:
I feel simply removing the `assert` is safer here. The problem is follower might have bigger `nextFileNumber` than leader.

Reply:
yeah, but that would change prod behavior.

Reply:
Actually, let's not, that would be a behavior change